### PR TITLE
Fix `-Werror`.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/compiler/diagnostic.h [hunk 1]
```diff
@@ -180,8 +185,8 @@ class NullDiagnostics : public Diagnostics {
   void end_group() {}
 
  protected:
-  void emit(Severity severity, const char* format, va_list& arguments) {}
-  void emit(Severity severity, Source::Range range, const char* format, va_list& arguments) {}
+  bool emit(Severity severity, const char* format, va_list& arguments) { return true; }
```
Comment (kasperl) [inline src/compiler/diagnostic.h:188]: Maybe add a comment to explain why returning true here is better than returning false?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/compiler/diagnostic.h (hunk 1 above)
Comment (floitsch) [reply src/compiler/diagnostic.h]: It doesn't really matter, but I agree "false" is better.

Comment (floitsch) [reply src/compiler/diagnostic.h:190]: Turns out, that we check whether we encountered ("printed") errors in asserts. So changing it back to "true".

**Pull request merged**